### PR TITLE
RXF OS Implementation

### PR DESCRIPTION
**Pull request by ArmenAg**
## What does this PR do?
Implements R3F and R4F coming from Facebook Research: https://arxiv.org/abs/2008.03156

This code was used to generate all the results from the paper excluding probing results.

### CODE REVIEW

File: fairseq/models/roberta/model.py [hunk 1]
```diff
@@ -98,6 +98,8 @@ def add_args(parser):
                             help='scalar quantization noise and scalar quantization at training time')
         parser.add_argument('--untie-weights-roberta', action='store_true',
                             help='Untie weights between embeddings and classifiers in RoBERTa')
+        parser.add_argument('--do-r4f', action='store_true',
```
Comment (myleott) [inline fairseq/models/roberta/model.py:101]: Please rename this to something more descriptive, something like `--spectral-norm-classification-head`

File: fairseq/models/bart/model.py [hunk 2]
```diff
@@ -59,21 +59,25 @@ def add_args(parser):
             choices=utils.get_available_activation_fns(),
             help='activation function to use for pooler layer'
         )
+        parser.add_argument('--do-r4f', action='store_true',
```
Comment (myleott) [inline fairseq/models/bart/model.py:62]: Please rename this to something more descriptive, something like `--spectral-norm-classification-head`

Comment (ArmenAg) [reply fairseq/models/bart/model.py]: Updated.

File: fairseq/models/roberta/model.py (hunk 1 above)
Comment (ArmenAg) [reply fairseq/models/roberta/model.py]: Updated.

**Pull request closed without merging**